### PR TITLE
fix: polish transforms

### PR DESCRIPTION
**Pull request by pfackeldey**
transforms now properly error also under jit transformation

## Summary by CodeRabbit

* **New Features**
  * Parameter transforms now accept array-like bounds and normalize numeric inputs consistently.

* **Bug Fixes**
  * Validation and error messages standardized and clarified to reference specific transform instances; stricter finite/positivity checks.

* **Tests**
  * Tests updated for new messages, broader exception coverage, and additional edge cases.

* **Chores**
  * Added a float-array normalization utility and declared new dependencies (jaxtyping, equinox).

### CODE REVIEW

File: src/everwillow/parameters/transforms.py [hunk 1]
```diff
@@ -220,15 +224,12 @@ class SoftPlusTransform(AbstractParameterTransformation):
...
     def unwrap(self, value: ArrayLike) -> ArrayLike:
         """Apply the inverse softplus, validating positivity and finiteness."""
-        value = jnp.asarray(value)
-        value = eqx.error_if(value, ~jnp.isfinite(value), "Value must be finite.")
+        value = float_array(value)
+        value = eqx.error_if(value, ~jnp.isfinite(value), "value must be finite.")
         value = eqx.error_if(
-            value, value < 0, "Expected positive inputs to inv_softplus."
+            value, value < 0, "expected positive inputs to inv_softplus."
         )
```
[the remark this answers is not in the document]
Comment (MoAly98) [reply src/everwillow/parameters/transforms.py]: @pfackeldey  i resolved this.
